index.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+/*global window :true*/
+
+// this 16777215 == ffffff in decimal
+const FFFFF = 16777215;
+// addapted from https://www.paulirish.com/2009/random-hex-color-code-snippets/
+const _getRandomColor = () => `#${Math.floor(Math.random() * FFFFF).toString(16)}`;
+
+const _isDomElement = (node = {}) => node.tagName;
+
+export class RandomColor {
+  constructor(repeate = false) {
+    this.state = {
+      repeate,
+      usedColors: [],
+    };
+  }
+
+  canRepeat() {
+    return this.state.repeate;
+  }
+
+  contains(color) {
+    return this.state.usedColors.indexOf(color) !== -1;
```

Comment:
the complexity of usedColors, would be O(n) given that the array is not sorted. 
to speed up this I can think in 2 options (to make the time complexity to O(1))

1) Changing `usedColors` for a `Set()`  (and we could use set.contain(color) later on)

2) Changing `usedColors` to an  Object with `[key]: value` where `key` would be the color and the `value` would just be a convenient boolean. (and we could use `!!this.state.useColors[color];`